packages/doenetml-worker-javascript/src/Core.js

```diff
@@ -3498,9 +3257,9 @@ export default class Core {
                 component,
                 varName: "isInactiveCompositeReplacement",
             });
-            for (let childIdx in component.allChildren) {
+            for (const childIdxStr in component.allChildren) {
                 await this.changeInactiveComponentAndDescendants(
-                    this._components[childIdx],
+                    this._components[childIdxStr],
```

Comment:
Does this have to be done sequentially? You could `.map` to a bunch of promises and do `await Promise.all`